index.js

```diff
@@ -30,6 +32,54 @@ module.exports = {
         "arrow-body-style": 0,
         "import/no-named-as-default": 0,
         "no-restricted-syntax": ["error", "ForInStatement", "LabeledStatement", "WithStatement"],
-        "no-prototype-builtins": 1
-    }
-};
+        "no-prototype-builtins": 1,
+
+        //Listed extensions are not linted on import
+        //
+        // imports cannot have .ts and .tsx extensions as it is default typescript behavior which cannot be changed at the moment
+        // and therefore it is listed so it will not be linted by eslint
+        //
+        // Correct: import Component from "path/to/Component.jsx"
+        // Correct: import jsFile from "path/to/jsFile"
+        // Wrong: import jsFile from "path/to/jsFile.js"
+        "import/extensions": [
+            "error",
+            "ignorePackages",
+            {
+                "js": "never",
+                "ts": "never",
+                "tsx": "never"
+            }
+        ]
+    },
+    "settings": {
+        "import/resolver": {
+            "typescript": {
+                "project": "./tsconfig.json"
+            }
+        }
+    },
+    "overrides": [
+        {
+            "files": ["**/*.ts", "**/*.tsx"],
+            "env": { "browser": true, "es6": true, "node": true },
+            "extends": [
+                "eslint:recommended",
+                "plugin:@typescript-eslint/eslint-recommended",
+                "plugin:@typescript-eslint/recommended"
+            ],
+            "parser": "@typescript-eslint/parser",
+            "parserOptions": {
+                "project": "./tsconfig.json"
+            },
+            "plugins": ["@typescript-eslint"],
+            "rules": {
+                "react/prop-types": "off",
```

Comment:
Could you add a comment that this is checked by typescript itself

Reply:
Updated